src/FakeItEasy/Core/ArgumentConstraintTrap.cs

```diff
@@ -18,7 +18,7 @@ internal class ArgumentConstraintTrap
             new ThreadLocal<Action<IArgumentConstraint>>(() => ThrowWhenUnpreparedToTrapConstraint);
 
         public static void ReportTrappedConstraint(IArgumentConstraint constraint) =>
-            saveTrappedConstraintAction.Value.Invoke(constraint);
+            saveTrappedConstraintAction.Value!.Invoke(constraint);

```

Comment:
Can't actually be null since a value factory is passed to the ThreadLocal constructor, and we never assign null to `Value`.